src/java.base/share/classes/java/net/Socket.java

```diff
@@ -319,6 +320,7 @@ private static Void checkPermission(SocketImpl impl) {
      * @see        java.net.SocketImpl
      * @see        SecurityManager#checkConnect
      */
+    @SuppressWarnings("this-escape")
```

Comment:
This is a weird one. I guess the issue here is that the escape happens in the chained constructor, and is propagated recursively up the constructor chain. Is the suppress warning here still needed after disabling this-escape warning at line 358?

Actually - these are all weird since the only place where the escape happens is in the private constructor at line 548 - and it doesn't even get flagged there (presumably because it's a private constructor?) 

I guess that the rationale is that subclasses cannot override the private constructor (where the escape happen), but can override the public constructor that calls the private constructor where the escape happen. I can't help feeling that the warning would be better place on the private constructor though. Seeing it here confused me a lot.